pkg/operators/kubeipresolver/kubeipresolver.go

```diff
@@ -226,72 +104,51 @@ func (m *KubeIPResolverInstance) PostGadgetRun() error {
 }
 
 func (m *KubeIPResolverInstance) enrich(ev any) {
-	additionalInfo, _ := ev.(KubeNetworkInformation)
-	containerInfo, _ := ev.(operators.ContainerInfoGetters)
+	additionalInfo, _ := ev.(KubeIPResolverInterface)
 
 	pods := m.manager.k8sInventory.GetPods()
 	foundLocal := false
 	foundRemote := 0
-	remoteIPs := additionalInfo.GetRemoteIPs()
-	endpoints := make([]types.EndpointDetails, len(remoteIPs))
+	endpoints := additionalInfo.GetEndpoints()
 	for j := range endpoints {
 		// initialize to this default value if we don't find a match
-		endpoints[j].Kind = types.RemoteKindOther
+		endpoints[j].Kind = types.EndpointKindRaw
 	}
 
-	for i, pod := range pods.Items {
-		if foundLocal && foundRemote == len(remoteIPs) {
+	for _, pod := range pods.Items {
+		if foundLocal && foundRemote == len(endpoints) {
 			break
 		}
-
-		if pod.Namespace == containerInfo.GetNamespace() && pod.Name == containerInfo.GetPod() {
-			foundLocal = true
-			owner := ""
-			// When the pod belongs to Deployment, ReplicaSet or DaemonSet, find the
-			// shorter name without the random suffix. That will be used to
-			// generate the network policy name.
-			if pods.Items[i].OwnerReferences != nil {
-				nameItems := strings.Split(pods.Items[i].Name, "-")
-				if len(nameItems) > 2 {
-					owner = strings.Join(nameItems[:len(nameItems)-2], "-")
-				}
-			}
-			additionalInfo.SetLocalPodDetails(owner, pod.Status.HostIP, pod.Status.PodIP, pod.Labels)
-		}
-
 		if pod.Spec.HostNetwork {
 			continue
 		}
 
-		for j, remoteIP := range remoteIPs {
-			if pod.Status.PodIP == remoteIP {
+		for j, endpoint := range endpoints {
+			if pod.Status.PodIP == endpoint.Addr {
 				foundRemote++
-				endpoints[j].Kind = types.RemoteKindPod
+				endpoints[j].Kind = types.EndpointKindPod
 				endpoints[j].Name = pod.Name
 				endpoints[j].Namespace = pod.Namespace
 				endpoints[j].PodLabels = pod.Labels
 			}
 		}
 	}
-	if foundRemote == len(remoteIPs) {
-		additionalInfo.SetEndpointsDetails(endpoints)
+	if foundRemote == len(endpoints) {
 		return
 	}
 
 	svcs := m.manager.k8sInventory.GetSvcs()
 
 	for _, svc := range svcs.Items {
```

Comment:
I think also here we should add a check to avoid iterating over all services:

```go
func (m *KubeIPResolverInstance) enrich(ev any) {
	additionalInfo, _ := ev.(KubeIPResolverInterface)
	endpoints := additionalInfo.GetEndpoints()
	for j := range endpoints {
		// initialize to this default value if we don't find a match
		endpoints[j].Kind = types.EndpointKindRaw
	}

	found := 0
	pods := m.manager.k8sInventory.GetPods()
	for _, pod := range pods.Items {
		if pod.Spec.HostNetwork {
			continue
		}

		for j, endpoint := range endpoints {
			if pod.Status.PodIP == endpoint.Addr {
				endpoints[j].Kind = types.EndpointKindPod
				endpoints[j].Name = pod.Name
				endpoints[j].Namespace = pod.Namespace
				endpoints[j].PodLabels = pod.Labels

				found++
				if found == len(endpoints) {
					break
				}
			}
		}
	}

	svcs := m.manager.k8sInventory.GetSvcs()
	for _, svc := range svcs.Items {
		for j, endpoint := range endpoints {
			if svc.Spec.ClusterIP == endpoint.Addr {
				endpoints[j].Kind = types.EndpointKindService
				endpoints[j].Name = svc.Name
				endpoints[j].Namespace = svc.Namespace
				endpoints[j].PodLabels = svc.Labels

				found++
				if found == len(endpoints) {
					break
				}
			}
		}
	}
}
```